### PR TITLE
ENT-10647: Adjusted hub install scripts to allow more time for postgres server to change state

### PR DESCRIPTION
**Pull request by craigcomstock**

### CODE REVIEW

**Comment by craigcomstock** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/9898/)

**Comment by craigcomstock** [comment]
squashed the commits, will try a rhel-7 hub package build first, manually test, if that looks OK I will start a full jenkins run.

I think this is fixed now. Primary cause of failures was a typo:

```
-          wait_for_cfpostgres_down || failure=1
+          wait_for_cf_postgres_down || failure=1
```

**Comment by craigcomstock** [comment]
@cf-bottom jenkins please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9903/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9903/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9903/

File: packaging/common/script-templates/script-common.sh [hunk 1]
```diff
@@ -96,25 +96,25 @@ wait_for_cf_postgres() {
...
 wait_for_cf_postgres_down() {
-    # wait for CFEngine Postgresql service to be shutdown, up to 5 sec.
+    # wait for CFEngine Postgresql service to be shutdown, up to 60 sec.
     # Returns 0 if postgresql service is not running
     # Returns non-0 otherwise (1 if exited by timeout)
-    for i in $(seq 1 5); do
+    for i in $(seq 1 60); do
         true "checking if Postgresql is shutdown..."
-        if ! "$PREFIX"/bin/pg_isready >/dev/null 2>&1; then
+        if ! cd /tmp && su cfpostgres -c "$PREFIX/bin/psql -l" >/dev/null 2>&1; then
```
Comment (nickanderson) [inline packaging/common/script-templates/script-common.sh:108]: Does this do the right thing if the psql command fails? I can't ever remember when you need to ()

**Review by nickanderson — APPROVED** [review]
I think it's ok, but unsure about compound commands in if

File: packaging/common/script-templates/script-common.sh (hunk 1 above)
Comment (craigcomstock) [reply packaging/common/script-templates/script-common.sh:108]: i did test it.

```
[root@rhel7 tmp]# cd /tmp && su cfpostgres -c "$PREFIX/bin/psql -l"                                         
                                                 List of databases                                          
    Name    |   Owner    | Encoding  | Collate | Ctype | ICU Locale | Locale Provider |      Access privileg
es                                                                                                          
------------+------------+-----------+---------+-------+------------+-----------------+---------------------
--------                                                                                                    
 cfdb       | cfpostgres | SQL_ASCII | C       | C     |            | libc            | =Tc/cfpostgres      
       +     

---


[root@rhel7 tmp]# echo $?
0
[root@rhel7 tmp]# cd /tmp && su cfpostgres -c "$PREFIX/bin/psql -l"^C
[root@rhel7 tmp]# systemctl stop cf-postgres
[root@rhel7 tmp]# cd /tmp && su cfpostgres -c "$PREFIX/bin/psql -l"
psql: error: connection to server on socket "/tmp/.s.PGSQL.5432" failed: No such file or directory
        Is the server running locally and accepting connections on that socket?
[root@rhel7 tmp]# echo $?
2
```

Comment (craigcomstock) [reply packaging/common/script-templates/script-common.sh:108]: ah, right, probably need to move the `!`

```
[root@rhel7 tmp]# ! cd /tmp && su cfpostgres -c "$PREFIX/bin/psql -l"
[root@rhel7 tmp]# echo $?
1
[root@rhel7 tmp]# ! cd /tmp && su cfpostgres -c "$PREFIX/bin/psql -l" && echo "yay"
[root@rhel7 tmp]# cd /tmp && ! su cfpostgres -c "$PREFIX/bin/psql -l" && echo "yay"
psql: error: connection to server on socket "/tmp/.s.PGSQL.5432" failed: No such file or directory
        Is the server running locally and accepting connections on that socket?
yay
```

Comment (craigcomstock) [reply packaging/common/script-templates/script-common.sh]: good catch, fixed! (and tested) :)

**Comment by craigcomstock** [comment]
prefer a PR which can build github workflows: https://github.com/cfengine/buildscripts/pull/1303

**Pull request closed without merging**